From be679872d17f9ff5405d0351866691788cec2f60 Mon Sep 17 00:00:00 2001
From: Trent Mick <trent.mick@joyent.com>
Date: Mon, 12 Dec 2016 12:18:15 -0800
Subject: [PATCH] joyent/node-manta#293 ~~ evaluation needs to be less pedantic
 Reviewed by: Brian Bennett <brian.bennett@joyent.com> Reviewed by: David
 Pacheco <dap@joyent.com> Approved by: David Pacheco <dap@joyent.com>

---
 CHANGES.md    | 2 +-
 lib/client.js | 4 +++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/CHANGES.md b/CHANGES.md
index cd699dd..8c4b6ee 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -2,7 +2,7 @@
 
 ## not yet released
 
-(nothing yet)
+- joyent/node-manta#293 '~~/' works, '~~' does now too.
 
 ## 4.1.0
 
diff --git a/lib/client.js b/lib/client.js
index c13be29..9fa1cea 100644
--- a/lib/client.js
+++ b/lib/client.js
@@ -282,7 +282,9 @@ function getPath(p, user, skipEncode) {
         user = null;
     }
 
-    if (/^~~\//.test(p)) {
+    if (p === '~~') {
+        p = '/' + (user || process.env.MANTA_USER);
+    } else if (/^~~\//.test(p)) {
         p = p.replace(/^~~\//, '/' + (user || process.env.MANTA_USER) + '/');
     }
 
-- 
2.21.0

